File: app/src/main/java/com/boost/espressotest/data/local/mapper/ProductContentMapper.java
package com.boost.espressotest.data.local.mapper;

import com.boost.espressotest.data.local.model.ProductContent;
import com.boost.espressotest.domain.model.Product;

import io.reactivex.functions.Function;

public class ProductContentMapper implements Function<ProductContent, Product> {

    @Override
    public Product apply(ProductContent productContent) {
        return new Product(
                productContent.getId(),
                productContent.getName(),
                productContent.getPriceInCents(),
                productContent.getProducerName(),
                productContent.getImageUrl(),
                productContent.isFavorite());
    }
}

File: app/src/main/java/com/boost/espressotest/data/local/mapper/ProductContentMapper.kt
package com.boost.espressotest.data.local.mapper

import com.boost.espressotest.data.local.model.ProductEntity
import com.boost.espressotest.domain.model.Product

import io.reactivex.functions.Function

class ProductContentMapper : Function<ProductEntity, Product> {

    override fun apply(productEntity: ProductEntity): Product {
        with(productEntity) {
            return Product(id, name, priceInCents, producerName, imageUrl, isFavorite)
        }
    }
}
